File: rentio-client/src/types/flat.ts
export interface Flat {
  id: number
  address: string
  city: string
  state: string
  numberOfRooms: number
  area: number
  rent: number
  description: string
  isAvailable: boolean
  averageRating?: number
  reviewCount?: number
}

export interface FlatFilters {
  city?: string
  state?: string
  minRent?: number
  maxRent?: number
  numberOfRooms?: number
  available?: boolean
  page?: number
  size?: number
  sort?: string
}

File: rentrix-server/src/main/java/com/rentrix/rentrixserver/entity/Flat.java
package com.rentrix.rentrixserver.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;

@Entity
@Table(name = "flats")
@Data
public class Flat {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String address;
	private String city;
	private String state;
	
	private Integer numberOfRooms;
	private BigDecimal area; // in square meters
	private BigDecimal rent; // in currency
	
	@Lob
	private String description;
	private Boolean available;
	
}